src/libxrpl/ledger/ApplyView.cpp

```diff
@@ -91,8 +91,16 @@ ApplyView::dirAdd(
         return page;
     }
 
+    // We rely on modulo arithmetic of unsigned integers (guaranteed in
+    // [basic.fundamental] paragraph 2) to detect page representation overflow.
+    // For signed integers this would be UB, hence static_assert here.
+    static_assert(std::is_unsigned_v<decltype(page)>);
```

Comment:
Would it be worth it to add a second check like `static_assert(std::numeric_limits<std::uint64_t>::max() + 1 == 0);`? Or would that just be redundant?